src/loaders/jobs/taskfarmer/taskfarmer_task_mgr.py

```diff
@@ -0,0 +1,316 @@
+            JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
+            * after the period of time, squeue command will return error code -1 and error message:
+            slurm_load_jobs error: Invalid job id specified
+
+        Matrix of job status:
+        Done file  Exit Code   Job listed    Job Status Code  Returned Status
+        Y          Any         N/A           N/A              COMPLETED
+        N          0           N             PD               PENDING
+        N          0           Y             R                RUNNING
+        N          -1          N/A           N/A              FAILED (If a job is cancelled, it will be treated as failed.)
+        N          0           N             N/A              FAILED (If a job is cancelled, it will be treated as failed.)
+        N          0           Y             CG               CANCELLED
+        """
+
+        # Once all tasks are completed, NERSC generates this file.
+        # (ref: https://docs.nersc.gov/jobs/workflow/taskfarmer/#output)
+        done_file = os.path.join(self.job_dir, NERSC_SLURM_DONE_FILE)
+        if os.path.isfile(done_file):
+            return JobStatus.COMPLETED
+
+        # check job status using squeue command
+        std_out_file, std_err_file, exit_code = tf_common.run_nersc_command(
+            ['squeue', '-j', str(job_id)], self.job_dir, log_file_prefix='squeue', check_return_code=False)
+
+        if exit_code != 0:
+            # job finished without creating the done file indicating the job is failed
+            return JobStatus.FAILED
+
+        with open(std_out_file, "r") as std_out, open(std_err_file, "r") as std_err:
+            squeue_out = std_out.read().strip()
+            if str(job_id) in squeue_out:
+                job_str = squeue_out.splitlines()[1]
+                parsed_status = job_str.split()[4]
+                if parsed_status == 'PD':
+                    return JobStatus.PENDING
+                elif parsed_status == 'R':
+                    return JobStatus.RUNNING
+                elif parsed_status == 'CG':
+                    return JobStatus.CANCELLED
+                else:
+                    raise ValueError(f"Unrecognized job status: {squeue_out}")
+            else:
+                return JobStatus.FAILED
+
+    def _append_task_info(self, task_info):
+        """
+        Append the task information to the task info file
+        """
+
+        if not all(key in task_info for key in REQUIRED_TASK_INFO_KEYS):
+            raise ValueError(f"task_info must contain all keys: {REQUIRED_TASK_INFO_KEYS}")
+
+        task_info.update({'kbase_collection': self.kbase_collection, 'load_ver': self.load_ver, 'tool': self.tool})
+
+        # Append the new record to the file
+        task_info_file = self._get_task_info_file()
+        with open(task_info_file, 'a') as writer:
+            fcntl.flock(writer.fileno(), fcntl.LOCK_EX)
+            json.dump(task_info, writer)
```

Comment:
out of curiosity why the switch from jsonlines?

Reply:
`writer.fileno()` doesn't work for` josnlines.open`

Reply:
:+1: